config: allow empty configuration file (#9846)
tarantool/tarantool: changelogs/unreleased/gh-9845-nil-error-on-empty-config.md

## bugfix/config

> * Fixed non-verbose error on empty config file. Now Tarantool can
>   successfully startup with empty config file using data from other
>   configuration sources (gh-9845).

Reviewer: ```suggestion
* Fixed a non-verbose error on an empty configuration file. Now Tarantool
  successfully starts up with an empty configuration file using data from other
  configuration sources (gh-9845).
```

Author: Thanks, applied.
